blender: 2.83.0 -> 2.83.1

r-ryantm opened the pull request:
Semi-automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/blender/versions.


meta.description for blender is: "3D Creation/Animation/Publishing System"

meta.homepage for blender is: "https://www.blender.org"

meta.changelog for blender is: ""


###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 2.83.1 with grep in /nix/store/yl41v8grdc244g50wvp4skngyd10qnyj-blender-2.83.1
- directory tree listing: https://gist.github.com/4c44063390d51a4f47bc7523e067391a

- du listing: https://gist.github.com/b9997de258017d4f8da32a5f821ec529


---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

1 package rebuild(s)

1 x86_64-linux rebuild(s)
0 i686-linux rebuild(s)
1 x86_64-darwin rebuild(s)
0 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
blender
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/yl41v8grdc244g50wvp4skngyd10qnyj-blender-2.83.1 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A blender https://github.com/r-ryantm/nixpkgs/archive/8ca6d3801ad85c989759f28270b43fd31ed12e8c.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/yl41v8grdc244g50wvp4skngyd10qnyj-blender-2.83.1
ls -la /nix/store/yl41v8grdc244g50wvp4skngyd10qnyj-blender-2.83.1/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- blender
</details>

---

###### Maintainer pings

cc @cillianderoiste @veprbl for testing.

r-ryantm committed:
blender: 2.83.0 -> 2.83.1

veprbl commented:
Changelog
=========

* Fix T77774: New undo code broken by ‘make local’ behavior
* Audio SDL: Video editor Sound muted without muting it
* Fix T77803: IK Degrees of freedom drawing glitch
* Fix (unreported) wrong size of UserPreferences’ `dupflag` parameter. Fix Applied rBdceaef92d7a4: Fix (unreported) wrong size of UserPreferences’ `dupflag` parameter
* Fix T77915: Cycles OSL microfacet closure not working in custom shaders
* GPencil: Fix unreported Vertex Opacity Overlay not working
* Fix T75414: Incorrect masking in Color Balance modifier
* Fix T76767: Cycles performance regression with CLI renders
* Fix T77853: Error reloading linked library
* Fix missing hinting information in monospace font, rB783d3c675ac2: Fix missing hinting information in default font
* Fix T77657: NVIDIA Quadro FX4800 crash on startup
* GPencil: Improve viewlayer masking check
* Fix T77148: Crash changing multiple values for sequencer strips
* Fix T77047: Dyntopo Sample detail size on hidden mesh causes crash
* Fix T74101: File Browser in macOS fullscreen crashes or makes windows unusable
* Fix T77504: Operator search gives wrong results
* Fix T76894: Disable clipping region selection in material/rendered mode
* Fix T72936: Incorrect gizmo orientation with inherit rotation disabled
* Fix menu operator/search clipping the last character Fix Applied rBf649e5c418d9: Fix menu operator/search clipping the last character
* Fix T77603: OSL parser fails when script ends with comment without newline
* Fix crash running “Edit Voxel Size” operator outside of a main 3D View region Fix Applied rB5fc252feaeab: Fix crash running “Edit Voxel Size” operator outside of a main 3D View region
* GPencil: Fix unreported error in Dots Strokes material initialization
* GPencil: Fix unreported Shift+F OPacity key not working
* Calculate epsilon values for interp_weights_poly to improve accuracy Fix Applied rB9d5e5e282cff: Calculate epsilon values for interp_weights_poly to improve accuracy
* Fix T76273 Glitches caused by glCopyImageSubData on windows + intel gpu Fix Applied rB7b754c8c9952: Fix T76273 Glitches caused by glCopyImageSubData on windows + intel gpu
* Fix T77358: Gpencil can’t select geometry within transparent layers
* Fix T77520: GPencil viewlayer filter produce crash with masking layers
* Fix T77367: Blender’s snap package ignores command line options
* Fix for T77095: work around render artifacts with AMD Radeon RX 4xx and 5xx
* Fix T77164: scaling/rotation fails for 3 selected NURB points
* Fix T77156: GPencil view layer filter by layer not working
* Fix T77456: Broken vertex paint undo on high-poly objects
* Fix T77448: Camera Solver constraint can’t be converted to f-curve

veprbl committed:
blender: don't require python3.7m on darwin

python is currently pointing to the python38, so this patches the
required version to match python version that is provided to the
expression. The m prefix denotes the pymalloc ABI for python 3.7 and
older, but is not provided starting python 3.8.

drewrisinger reviewed (approved):
* [x] Diff LGTM
* [x] Building: Attempted to build for ~10 mins on my PC, but it was maxing out RAM & had to kill. Got ~10% through build process.
* [x] OfBorg: I think it approves? Looks like one run succeeded, other timed out.

jollheef reviewed (approved):
- [x] LGTM
- [x] Tested on NixOS
